code/game/objects/items/airlock_painter.dm

```diff
@@ -41,11 +46,28 @@
 	. = ..()
 	ink = new initial_ink_type(src)
 
+/obj/item/airlock_painter/update_icon()
+	var/base = initial(icon_state)
+	if(!istype(ink))
+		icon_state = "[base]_none"
+		return
+	switch(ink.charges/ink.max_charges)
+		if(0.001 to PAINTER_LOW)
+			icon_state = "[base]_low"
```

Comment:
instead of changing the icon_state and also adding new icons for each version, just make overlays and use add_overlay

Reply:
every..other..sprite change for low power does this (look at welding tool) i dont know how to useb overlays at all its FINE its FINE it wORKS its FINE ok**

Reply:
cut_overlays()
add_overlay(mutable_appearance(icon, 'some_iconstate'))

done

Reply:
woah, it isn't personal, just making sure things are in the best practice and most maintainable form while we can - lots of things work but can still not be ideal. If it's that much of an issue I suppose it can go without it, but please don't take the reviews personally.